modules/compute/gke-node-pool/variables.tf

```diff
@@ -391,3 +391,25 @@ variable "max_pods_per_node" {
   type        = number
   default     = null
 }
+
+variable "upgrade_settings" {
+  description = "define node pool upgrade settings"
+  type = object({
+    strategy        = string,
+    max_surge       = optional(number),
+    max_unavailable = optional(number)
+  })
+  default = {
+    strategy        = "SURGE",
+    max_surge       = 0,
+    max_unavailable = 1
+  }
+  validation {
+    condition = (
+      contains(["SURGE"], var.upgrade_settings.strategy) &&
+      var.upgrade_settings.max_unavailable >= 0 &&
+      var.upgrade_settings.max_surge >= 0 &&
+    (var.upgrade_settings.max_unavailable > 0 || var.upgrade_settings.max_surge > 0))
+    error_message = "upgrade settings are invalid"
```

Comment:
Please throw specific, actionable error messages when possible. One error message per and operand would be nice here

Reply:
The issue with one error message per "and" is that user will have to figure out which one is causing the failure. Should we instead link the terraform doc instead? 

Proposed error_message:
"Only surge strategy is supported. Refer to https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/container_node_pool#upgrade_settings-1 for more details"